sw/device/lib/flash_ctrl.c

```diff
@@ -113,10 +115,41 @@ static int flash_write_internal(uint32_t addr, part_type_t part,
   return get_clr_err();
 }
 
+// address is assumed to be aligned to uint32_t
 int flash_write(uint32_t addr, part_type_t part, const uint32_t *data,
                 uint32_t size) {
-  // TODO: Breakdown into FIFO chunks if needed.
-  return flash_write_internal(addr, part, data, size);
+  uint32_t window_offset = (addr / sizeof(uint32_t)) % PROGRAM_RESOLUTION_WORDS;
+  uint32_t max_words = PROGRAM_RESOLUTION_WORDS;
+
+  // if initial address isn't aligned, the delta is the max
+  // number of words that can be programmed.
+  if (window_offset > 0) {
+    max_words = PROGRAM_RESOLUTION_WORDS - window_offset;
```

Comment:
This if isn't actually necessary, and you can just do `uint32_t max_words = PROGRAM_RESOLUTION_WORDS - window_offset;`, since it's the same as above if window_offset = 0.

Reply:
o good point